src/script/script.cpp

```diff
@@ -244,24 +245,45 @@ bool CScript::IsPayToScriptHash() const
     // Extra-fast test for pay-to-script-hash CScripts:
 
     // Check if this script is P2SH without OP_CHECKBLOCKATHEIGHT
-    bool p2sh = (this->size() == 23 &&
-                 this->at(0) == OP_HASH160 &&
-                 this->at(1) == 0x14 &&
-                 this->at(22) == OP_EQUAL);
+    bool p2sh = (size() == 23 &&
+                 at(0) == OP_HASH160 &&
+                 at(1) == 0x14 &&
+                 at(22) == OP_EQUAL);
 
     // Check if this script is P2SH with OP_CHECKBLOCKATHEIGHT
     // The overall size should not be more then 62:
     // 23 bytes for common P2SH + 1 byte size + 32 bytes of block hash + 1 byte size + 4 bytes of block height + 1 byte opcode
-    bool p2shWithReplay = (this->size() > 23 &&
-                           this->size() < 63 &&
-                           this->at(0) == OP_HASH160 &&
-                           this->at(1) == 0x14 &&
-                           this->at(22) == OP_EQUAL &&
-                           this->at(this->size() - 1) == OP_CHECKBLOCKATHEIGHT);
+    bool p2shWithReplay = (size() > 23 &&
+                           size() < 63 &&
+                           at(0) == OP_HASH160 &&
+                           at(1) == 0x14 &&
+                           at(22) == OP_EQUAL &&
+                           at(size() - 1) == OP_CHECKBLOCKATHEIGHT);
 
     return p2sh || p2shWithReplay;
 }
 
+bool CScript::IsPayToPublicKey() const
+{
+    // Extra-fast test for pay-to-pubkey CScripts:
+
+    // Check if this script is P2PK without OP_CHECKBLOCKATHEIGHT
+    bool p2pk = (size() == 35 &&
+                 at(0) == 0x21 &&
+                 at(34) == OP_CHECKSIG);
+
+     // Check if this script is P2PKH with OP_CHECKBLOCKATHEIGHT
+     // The overall size should not be more then 74:
```

Comment:
This should be P2PK, right?
```suggestion
     // Check if this script is P2PK with OP_CHECKBLOCKATHEIGHT
     // The overall size should not be more than 74:
```

Reply:
Correct, done